### ISSUE TITLE
accessibility.bsh unit test fails

### ISSUE THREAD

**Issue by GoogleCodeExporter** [pejobo/beanshell2#4]
```
Looking at the code, I don't agree with the failed assertion:

assert(m.foo(null).equals("public"));

m.foo(null) is ambiguous and BeanShell takes the first matching method.  Is
that acceptable?  If yes, then we should remove this assertion.

Here is a simpler version of the same issue:

in T.java:

public class T {
    public static String foo( Integer x ) { return "integer"; }
    public static String foo( String x ) { return "string"; }
}

Then try this BeanShell:

print(T.foo(null));

What should happen?

```

Original issue reported on code.google.com by `fschm...@gmail.com` on 5 May 2007 at 8:22
